core/src/main/groovy/com/novoda/gradle/release/AndroidLibrary.groovy

```diff
@@ -13,60 +14,61 @@ import org.gradle.api.model.ObjectFactory
 
 class AndroidLibrary implements SoftwareComponentInternal {
 
-    private final UsageContext runtimeUsage
+    private final Set<UsageContext> usages
 
-    public static AndroidLibrary newInstance(Project project) {
+    AndroidLibrary(Project project) {
+        this.usages = new DefaultDomainObjectSet<UsageContext>(UsageContext)
 
-        ObjectFactory objectFactory = project.getObjects();
-        Usage usage = objectFactory.named(Usage.class, Usage.JAVA_RUNTIME);
+        ObjectFactory objectFactory = project.getObjects()
+        Usage api = objectFactory.named(Usage.class, Usage.JAVA_API)
+        Usage runtime = objectFactory.named(Usage.class, Usage.JAVA_RUNTIME)
 
-
-        def configuration = project.configurations.getByName("compile")
-        return configuration ? from(configuration, usage) : empty()
-    }
-
-    static AndroidLibrary from(def configuration, Usage usage) {
-        def runtimeUsage = new RuntimeUsage(configuration.dependencies, usage)
-        new AndroidLibrary(runtimeUsage)
-    }
-
-    static AndroidLibrary empty() {
-        def usage = new RuntimeUsage(new DefaultDomainObjectSet<Dependency>(Dependency))
-        new AndroidLibrary(usage)
+        addUsageContextFromConfiguration(project, "compile", api)
+        addUsageContextFromConfiguration(project, "api", api)
+        addUsageContextFromConfiguration(project, "implementation", runtime)
     }
 
-    AndroidLibrary(UsageContext runtimeUsage) {
-        this.runtimeUsage = runtimeUsage
+    String getName() {
+        return "android"
     }
 
-    public String getName() {
-        return "android"
+    Set<UsageContext> getUsages() {
+        return usages
```

Comment:
Please keep the Collections.singleton()... Or is there anything which prevent us to use it?

Reply:
As this will support multiple usages, that's why I just return the usage list directly.